fjord/feedback/views.py

```diff
@@ -166,23 +166,29 @@ def _handle_feedback_post(request, locale=None, product=None,
     # to do so.
     if request.BROWSER != UNKNOWN:
         # If we don't have a product, try to infer that from the user
-        # agent.
-        if not product:
-            product = models.Response.infer_product(request.BROWSER)
+        # agent information.
+        if not product_db_name:
+            product_db_name = models.Response.infer_product(request.BROWSER)
 
-        # If we have a product and it matches the user agent product,
+        # If we have a product and it matches the user agent browser,
         # then we can infer the version and platform from the user
         # agent if they're missing.
-        if (product and product == models.Response.infer_product(
-                request.BROWSER)):
-            if not version:
-                version = request.BROWSER.browser_version
-            if not platform:
-                platform = models.Response.infer_platform(
-                    product, request.BROWSER)
+        if product_db_name:
+            product = models.Product.objects.get(db_name=product_db_name)
+            # So instead, we do the first test to handle the Android
+            # case and the second test to handle the Firefox Dev and
+            # related cases.
```

Comment:
I think you can remove this comment and the commented out code below? Or perhaps I am confused.